src/context/Iterator.cpp

```diff
@@ -589,6 +616,15 @@ List PropIter::getEdges() {
     return edges;
 }
 
+const Value& PropIter::getColumn(int32_t index) const {
+    auto size = iter_->size();
+    if (static_cast<size_t>(std::abs(index)) >= size) {
+        return Value::kNullBadType;
+    }
+    auto currentRow = *iter_;
+    return currentRow[(size + index) % size];
+}
```

Comment:
Pull up this implementation to base class for code reusing.

Reply:
ok